### DISCUSSION TITLE
Direct Import components without folder-based naming

### DISCUSSION THREAD

**Discussion by millansingh** [nuxt/nuxt#17583] [Questions] [answered]
Hi there.

I've been trying to migrate my first project to Nuxt 3. I've used Nuxt 2 in the past, and I was able to directly import my components in Nuxt 2. For instance, if I had a component in "components/atoms/Badge.vue" I could import with "import Badge from '~/components/atoms/Badge'".

I am failing to achieve a similar outcome with Nuxt 3. I can get an import to work using the folder-component name -- atomsBadge -- but this looks silly to me, and I'd like to not have to use this naming technique/auto-import. As the Nuxt 3 docs are still incomplete, I haven't been able to find a sufficient answer there or on Google, so I've come here to ask you fine folks instead.

Thanks!

**Comment by TheAlexLichter** [comment, marked as answer]
You mean importing components "manually" as below? This should work fine ☺️ 

```vue
<script setup>
import Badge from "~/components/atoms/Badge.vue";
</script>

<template>
  <div>
    <Badge />
  </div>
</template>
```

[Example sandbox](https://codesandbox.io/s/long-tdd-ec4u7g?file=/app.vue:0-130)
